### PR TITLE
Unescape Type.Name & Type.FullName, fix #370

### LINKED ISSUE #370: Import(System.Type) should unescape

### PR DESCRIPTION
**Pull request by CreateAndInject**
No need to unescape Type.Namespace.
This PR can fix #370

### CODE REVIEW

File: src/DotNet/ReflectionExtensions.cs [hunk 1]
```diff
@@ -105,5 +105,11 @@ static class ReflectionExtensions {
 		/// <param name="type">this</param>
 		public static bool IsTypeDef(this Type type) =>
 			!(type is null) && !type.HasElementType && (!type.IsGenericType || type.IsGenericTypeDefinition);
+
+		internal static string Unescape(string name) {
+			if (string.IsNullOrEmpty(name))
+				return name;
+			return name.Replace(@"\,", ",").Replace(@"\+", "+").Replace(@"\&", "&").Replace(@"\*", "*").Replace(@"\[", "[").Replace(@"\]", "]").Replace(@"\\", @"\");
```
Comment (0xd4d) [inline src/DotNet/ReflectionExtensions.cs:112]: This code is inefficient. You should have a check to see if it has a `\` and if not, return the input string. If there's a `\`, use a string builder.

File: src/DotNet/ReflectionExtensions.cs [hunk 2]
```diff
@@ -105,5 +106,33 @@ static class ReflectionExtensions {
 		/// <param name="type">this</param>
 		public static bool IsTypeDef(this Type type) =>
 			!(type is null) && !type.HasElementType && (!type.IsGenericType || type.IsGenericTypeDefinition);
+
+		internal static string Unescape(string name) {
+			if (string.IsNullOrEmpty(name) || !name.Contains(@"\"))
```
Comment (0xd4d) [inline src/DotNet/ReflectionExtensions.cs:111]: `name.IndexOf('\\') < 0` is better since it just searches for one char instead of a substring.

**Comment by CreateAndInject** [comment]
> Are method/field/prop/event names also escaped?

No, Reflection doesn't escape these names.

**Comment by 0xd4d** [comment]
Thanks, merged!

**Pull request merged**